internal/wasm/store_module_list.go

```diff
@@ -44,7 +47,11 @@ func (s *Store) deleteModule(moduleName string) error {
 	if node.next != nil {
 		node.next.prev = node.prev
 	}
-	delete(s.nameToNode, moduleName)
+
+	if node.name != "" {
+		delete(s.nameToNode, node.name)
+	}
```

Comment:
This could move in the first block where we check `if node.name != ""` so we don't have to do it twice.